### PR TITLE
Session expiration handling, NLMTA lookup error catching, security vulnerability patching

### PR DESCRIPTION
**Pull request by cameronblandford**
* #345 
* #664 
* #688 
* #689

### CODE REVIEW

**Comment by karenhanson** [comment]
I currently see the following:

```
HEAD is now at 4b523ee... Pull in upstream develop
npm WARN notice [SECURITY] lodash has the following vulnerability: 1 low. Go here for more details: https://nodesecurity.io/advisories?search=lodash&version=3.10.1 - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
npm WARN notice [SECURITY] hoek has the following vulnerability: 1 moderate. Go here for more details: https://nodesecurity.io/advisories?search=hoek&version=2.16.3 - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
```

**Comment by cameronblandford** [comment]
@karenhanson there's no clean or easy way to get rid of those. These are sub-sub-sub-sub-...-sub dependencies of the most recent versions of necessary packages, sadly. Luckily, they're minor vulnerabilities that can't be used to accomplish much - the linked pages describe what exactly they do, and `npm audit` can show you exactly how deeply those deps are nested within the dependency tree. The only good solution is to wait until new versions of the relevant top-level packages come out.

File: app/components/workflow-component.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import Component from '@ember/component';
+import ENV from '../config/environment';
+export default Component.extend({
+  toast: Ember.inject.service('toast'),
```
Comment (karenhanson) [inline app/components/workflow-component.js]: is toast still needed here?

**Comment by karenhanson** [comment]
@cameronblandford confirmed by @htpvu - a failure to get a response from nlmta should log a line to the console.log, and otherwise fail quietly for the user - no popup message, just allow the code to continue without failing.

**Pull request merged**